akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/kubernetes/ClusterServiceDiscovery.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017 Lightbend, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lightbend.rp.akkaclusterbootstrap.kubernetes
+
+import akka.actor.ActorSystem
+import akka.discovery.ServiceDiscovery
+import akka.http.scaladsl._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{ Authorization, OAuth2BearerToken }
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.FileIO
+import com.lightbend.rp.akkaclusterbootstrap.AkkaManagementPortName
+import com.lightbend.rp.common._
+import com.typesafe.sslconfig.akka.AkkaSSLConfig
+import com.typesafe.sslconfig.ssl.TrustStoreConfig
+import java.nio.file.Paths
+import scala.collection.immutable.Seq
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Try
+
+import JsonFormat._
+import ServiceDiscovery._
+
+object ClusterServiceDiscovery {
+  private[kubernetes] def targets(podList: PodList, name: String): Seq[ResolvedTarget] =
+    for {
+      item <- podList.items
+      container <- item.spec.containers.find(_.name == name)
+      port <- container.ports.find(_.name == AkkaManagementPortName)
+      ip <- item.status.podIP
+    } yield ResolvedTarget(ip, Some(port.containerPort))
+}
+
+/**
+ * An alternative implementation that uses the Kubernetes API. Currently,
+ * reactive-cli specifies this via arguments when it generates resources
+ * for Kubernets targets. Thus, the default implementation for reactive-lib
+ * remains one based on the service locator.
+ *
+ * The main advantage of this is that we can discover pods that aren't yet ready,
+ * perhaps they have health checks that rely on Akka Cluster being up.
+ */
+class ClusterServiceDiscovery(system: ActorSystem) extends ServiceDiscovery {
```

Comment:
Once I'm done refactoring things today, this would be nice to add into akka-management directly, so people can select it if they wanted to

Reply:
Good idea - I'd be happy to submit a PR once the modularization is ready.